From 4a32b73dd75fcc955f220eccc8f70ea429dd3eb4 Mon Sep 17 00:00:00 2001
From: Tzvetan Mikov <tmikov@fb.com>
Date: Mon, 8 Apr 2019 19:08:40 -0700
Subject: [PATCH] Disable setting of the global new handler.

---
 lib/Support/ErrorHandling.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/lib/Support/ErrorHandling.cpp b/lib/Support/ErrorHandling.cpp
index cb14749cc42..6e85d824c0d 100644
--- a/lib/Support/ErrorHandling.cpp
+++ b/lib/Support/ErrorHandling.cpp
@@ -198,6 +198,8 @@ void llvm::install_out_of_memory_new_handler() {
   }
 }
 
+// T42835532: LLVM shouldn't be changing the global new handler by default.
+#if 0
 // Static object that causes installation of 'out_of_memory_new_handler' before
 // execution of 'main'.
 static class NewHandlerInstaller {
@@ -207,6 +209,7 @@ public:
   }
 } new_handler_installer;
 #endif
+#endif
 
 void llvm::llvm_unreachable_internal(const char *msg, const char *file,
                                      unsigned line) {
-- 
2.13.5

