--- a/lib/Support/Signals.cpp
+++ b/lib/Support/Signals.cpp
@@ -36,10 +36,14 @@
 
 using namespace llvm;
 
+#if 0
 static cl::opt<bool>
     DisableSymbolication("disable-symbolication",
                          cl::desc("Disable symbolizing crash backtraces."),
                          cl::init(false), cl::Hidden);
+#else
+const bool DisableSymbolication = true;
+#endif
 
 static ManagedStatic<std::vector<std::pair<void (*)(void *), void *>>>
     CallBacksToRun;
