--- StringExtras.h	2019-02-20 10:13:14.923803141 -0800
+++ StringExtras.h.new	2019-02-19 10:16:40.233342263 -0800
@@ -186,9 +186,11 @@
   return detail::to_float(T, Num, strtod);
 }
 
+#ifndef __ANDROID__
 inline bool to_float(const Twine &T, long double &Num) {
   return detail::to_float(T, Num, strtold);
 }
+#endif
 
 inline std::string utostr(uint64_t X, bool isNeg = false) {
   char Buffer[21];
