diff --git a/ReactAndroid/build.gradle b/ReactAndroid/build.gradle
index fd03e602cf..f6e7f1a675 100644
--- a/ReactAndroid/build.gradle
+++ b/ReactAndroid/build.gradle
@@ -160,6 +160,17 @@ task prepareJSC(dependsOn: downloadJSC) << {
     }
 }
 
+task prepareHermes() << {
+    def hermesAAR = new File("../../../../build_android/outputs/aar/android.aar")
+    def soFiles = zipTree(hermesAAR).matching({ it.include "**/*.so" })
+
+    copy {
+        from soFiles
+        from "../../../../hermes/first-party/libhermes/Android.mk"
+        into "$thirdPartyNdkDir/hermes"
+    }
+}
+
 task downloadNdkBuildDependencies {
     if (!boostPath) {
         dependsOn downloadBoost
@@ -221,7 +232,7 @@ def getNdkBuildFullPath() {
     return ndkBuildFullPath
 }
 
-task buildReactNdkLib(dependsOn: [prepareJSC, prepareBoost, prepareDoubleConversion, prepareFolly, prepareGlog], type: Exec) {
+task buildReactNdkLib(dependsOn: [prepareJSC, prepareHermes, prepareBoost, prepareDoubleConversion, prepareFolly, prepareGlog], type: Exec) {
     inputs.dir('src/main/jni/react')
     outputs.dir("$buildDir/react-ndk/all")
     commandLine getNdkBuildFullPath(),
@@ -288,7 +299,7 @@ android {
         jniLibs.srcDir "$buildDir/react-ndk/exported"
         res.srcDirs = ['src/main/res/devsupport', 'src/main/res/shell', 'src/main/res/views/modal', 'src/main/res/views/uimanager']
         java {
-            srcDirs = ['src/main/java', 'src/main/libraries/soloader/java', 'src/main/jni/first-party/fb/jni/java']
+            srcDirs = ['src/main/java', 'src/main/libraries/soloader/java', 'src/main/jni/first-party/fb/jni/java', '../../../../hermes/first-party/hermes-executor/src/main/java']
             exclude 'com/facebook/react/processing'
             exclude 'com/facebook/react/module/processing'
         }
diff --git a/ReactAndroid/src/main/jni/Application.mk b/ReactAndroid/src/main/jni/Application.mk
index c73c0397f5..62321559b9 100644
--- a/ReactAndroid/src/main/jni/Application.mk
+++ b/ReactAndroid/src/main/jni/Application.mk
@@ -32,3 +32,4 @@ APP_CPPFLAGS := -std=c++1y
 APP_LDFLAGS := -Wl,--build-id
 
 NDK_TOOLCHAIN_VERSION := clang
+HERMES_PATH := $(call my-dir)/../../../../../../../hermes
diff --git a/ReactAndroid/src/main/jni/react/jni/Android.mk b/ReactAndroid/src/main/jni/react/jni/Android.mk
index 9259233cca..78e8804156 100644
--- a/ReactAndroid/src/main/jni/react/jni/Android.mk
+++ b/ReactAndroid/src/main/jni/react/jni/Android.mk
@@ -54,6 +54,7 @@ include $(BUILD_SHARED_LIBRARY)
 $(call import-module,folly)
 $(call import-module,fb)
 $(call import-module,jsc)
+$(call import-module,hermes)
 $(call import-module,fbgloginit)
 $(call import-module,yogajni)
 $(call import-module,cxxreact)
@@ -65,3 +66,4 @@ $(call import-module,jsiexecutor)
 # $(call import-module,jscexecutor)
 
 include $(REACT_SRC_DIR)/jscexecutor/Android.mk
+include $(HERMES_PATH)/first-party/hermes-executor/src/main/java/com/facebook/hermes/reactexecutor/Android.mk
diff --git a/ReactCommon/jsi/Android.mk b/ReactCommon/jsi/Android.mk
index 4e650d9ca3..db81fe24ad 100644
--- a/ReactCommon/jsi/Android.mk
+++ b/ReactCommon/jsi/Android.mk
@@ -9,9 +9,9 @@ include $(CLEAR_VARS)
 
 LOCAL_MODULE := jsi
 
-LOCAL_SRC_FILES := $(wildcard $(LOCAL_PATH)/*.cpp)
+LOCAL_SRC_FILES := $(wildcard $(LOCAL_PATH)/*.cpp) $(wildcard $(HERMES_PATH)/API/jsi/jsi/adapter.cpp $(HERMES_PATH)/../jsi/jsi/adapter.cpp)
 
-LOCAL_C_INCLUDES += $(LOCAL_PATH)/..
+LOCAL_C_INCLUDES += $(LOCAL_PATH)/.. $(wildcard $(HERMES_PATH)/API/jsi $(HERMES_PATH)/../jsi)
 LOCAL_EXPORT_C_INCLUDES := $(LOCAL_C_INCLUDES)
 
 LOCAL_CFLAGS := -fexceptions -frtti -O3
