diff --git a/ReactCommon/jsi/JSCRuntime.cpp b/ReactCommon/jsi/JSCRuntime.cpp
index 3e9543653a..b8bd316c20 100644
--- a/ReactCommon/jsi/JSCRuntime.cpp
+++ b/ReactCommon/jsi/JSCRuntime.cpp
@@ -37,8 +37,21 @@ class JSCRuntime : public jsi::Runtime {
   ~JSCRuntime();
 
   void evaluateJavaScript(
-      std::unique_ptr<const jsi::Buffer> buffer,
+      const std::shared_ptr<const jsi::Buffer>& buffer,
       const std::string& sourceURL) override;
+
+  std::shared_ptr<const jsi::PreparedJavaScript> prepareJavaScript(
+      const std::shared_ptr<const jsi::Buffer>& buffer,
+      std::string sourceURL) override {
+    assert(false && "Not implemented");
+		return nullptr;
+	}
+
+  void evaluatePreparedJavaScript(
+      const std::shared_ptr<const jsi::PreparedJavaScript>& js) override {
+    assert(false && "Not implemented");
+  }
+
   jsi::Object global() override;
 
   std::string description() override;
@@ -315,7 +328,7 @@ JSCRuntime::~JSCRuntime() {
 }
 
 void JSCRuntime::evaluateJavaScript(
-    std::unique_ptr<const jsi::Buffer> buffer,
+    const std::shared_ptr<const jsi::Buffer>& buffer,
     const std::string& sourceURL) {
   std::string tmp(
       reinterpret_cast<const char*>(buffer->data()), buffer->size());
