diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3e40d7e087a..9719f8c8fc8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -803,7 +803,7 @@ include_directories( ${LLVM_INCLUDE_DIR} ${LLVM_MAIN_INCLUDE_DIR})

 # when crosscompiling import the executable targets from a file
 if(LLVM_USE_HOST_TOOLS)
-  include(CrossCompile)
+  # include(CrossCompile)
 endif(LLVM_USE_HOST_TOOLS)
 if(LLVM_TARGET_IS_CROSSCOMPILE_HOST)
 # Dummy use to avoid CMake Warning: Manually-specified variables were not used
diff --git a/cmake/modules/HandleLLVMOptions.cmake b/cmake/modules/HandleLLVMOptions.cmake
index 86ce0bbcfd0..5355b06f697 100644
--- a/cmake/modules/HandleLLVMOptions.cmake
+++ b/cmake/modules/HandleLLVMOptions.cmake
@@ -243,7 +243,7 @@ if( CMAKE_SIZEOF_VOID_P EQUAL 8 AND NOT WIN32 )

     # FIXME: CMAKE_SIZEOF_VOID_P is still 8
     add_definitions(-D_LARGEFILE_SOURCE)
-    add_definitions(-D_FILE_OFFSET_BITS=64)
+    # add_definitions(-D_FILE_OFFSET_BITS=64)
   endif( LLVM_BUILD_32_BITS )
 endif( CMAKE_SIZEOF_VOID_P EQUAL 8 AND NOT WIN32 )

diff --git a/tools/llvm-config/CMakeLists.txt b/tools/llvm-config/CMakeLists.txt
index f59402ac4b0..8b20cec52f3 100644
--- a/tools/llvm-config/CMakeLists.txt
+++ b/tools/llvm-config/CMakeLists.txt
@@ -65,17 +65,17 @@ endif()
 # Add the dependency on the generation step.
 add_file_dependencies(${CMAKE_CURRENT_SOURCE_DIR}/llvm-config.cpp ${BUILDVARIABLES_OBJPATH})

-if(CMAKE_CROSSCOMPILING AND NOT LLVM_CONFIG_PATH)
-  set(LLVM_CONFIG_PATH "${LLVM_NATIVE_BUILD}/bin/llvm-config" CACHE STRING "")
-
-  add_custom_command(OUTPUT "${LLVM_CONFIG_PATH}"
-    COMMAND ${CMAKE_COMMAND} --build . --target llvm-config --config $<CONFIGURATION>
-    DEPENDS ${LLVM_NATIVE_BUILD}/CMakeCache.txt
-    WORKING_DIRECTORY ${LLVM_NATIVE_BUILD}
-    COMMENT "Building native llvm-config..."
-    USES_TERMINAL)
-  add_custom_target(NativeLLVMConfig DEPENDS ${LLVM_CONFIG_PATH})
-  add_dependencies(NativeLLVMConfig CONFIGURE_LLVM_NATIVE)
-
-  add_dependencies(llvm-config NativeLLVMConfig)
-endif()
+# if(CMAKE_CROSSCOMPILING AND NOT LLVM_CONFIG_PATH)
+#   set(LLVM_CONFIG_PATH "${LLVM_NATIVE_BUILD}/bin/llvm-config" CACHE STRING "")
+#
+#   add_custom_command(OUTPUT "${LLVM_CONFIG_PATH}"
+#     COMMAND ${CMAKE_COMMAND} --build . --target llvm-config --config $<CONFIGURATION>
+#     DEPENDS ${LLVM_NATIVE_BUILD}/CMakeCache.txt
+#     WORKING_DIRECTORY ${LLVM_NATIVE_BUILD}
+#     COMMENT "Building native llvm-config..."
+#     USES_TERMINAL)
+#   add_custom_target(NativeLLVMConfig DEPENDS ${LLVM_CONFIG_PATH})
+#   add_dependencies(NativeLLVMConfig CONFIGURE_LLVM_NATIVE)
+#
+#   add_dependencies(llvm-config NativeLLVMConfig)
+# endif()
