diff --git a/CMakeLists.txt b/CMakeLists.txt
index a5b96569f9c..b25336708f0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -784,7 +784,7 @@ include_directories( ${LLVM_INCLUDE_DIR} ${LLVM_MAIN_INCLUDE_DIR})
 
 # when crosscompiling import the executable targets from a file
 if(LLVM_USE_HOST_TOOLS)
-  include(CrossCompile)
+    #  include(CrossCompile)
 endif(LLVM_USE_HOST_TOOLS)
 if(LLVM_TARGET_IS_CROSSCOMPILE_HOST)
 # Dummy use to avoid CMake Wraning: Manually-specified variables were not used
diff --git a/cmake/modules/HandleLLVMOptions.cmake b/cmake/modules/HandleLLVMOptions.cmake
index d15933157d2..9c3abce76bc 100644
--- a/cmake/modules/HandleLLVMOptions.cmake
+++ b/cmake/modules/HandleLLVMOptions.cmake
@@ -236,7 +236,7 @@ endif( CMAKE_SIZEOF_VOID_P EQUAL 8 AND NOT WIN32 )
 # so that off_t can stored offset > 2GB
 if( CMAKE_SIZEOF_VOID_P EQUAL 4 )
   add_definitions( -D_LARGEFILE_SOURCE )
-  add_definitions( -D_FILE_OFFSET_BITS=64 )
+  #  add_definitions( -D_FILE_OFFSET_BITS=64 )
 endif()
 
 if( XCODE )
diff --git a/tools/llvm-config/CMakeLists.txt b/tools/llvm-config/CMakeLists.txt
index 5112648ea73..51dce19fd14 100644
--- a/tools/llvm-config/CMakeLists.txt
+++ b/tools/llvm-config/CMakeLists.txt
@@ -62,18 +62,18 @@ add_definitions(-DCMAKE_CFG_INTDIR="${CMAKE_CFG_INTDIR}")
 # Add the dependency on the generation step.
 add_file_dependencies(${CMAKE_CURRENT_SOURCE_DIR}/llvm-config.cpp ${BUILDVARIABLES_OBJPATH})
 
-if(CMAKE_CROSSCOMPILING)
-  set(${project}_LLVM_CONFIG_EXE "${LLVM_NATIVE_BUILD}/bin/llvm-config")
-  set(${project}_LLVM_CONFIG_EXE ${${project}_LLVM_CONFIG_EXE} PARENT_SCOPE)
-
-  add_custom_command(OUTPUT "${${project}_LLVM_CONFIG_EXE}"
-    COMMAND ${CMAKE_COMMAND} --build . --target llvm-config --config $<CONFIGURATION>
-    DEPENDS ${LLVM_NATIVE_BUILD}/CMakeCache.txt
-    WORKING_DIRECTORY ${LLVM_NATIVE_BUILD}
-    COMMENT "Building native llvm-config..."
-    USES_TERMINAL)
-  add_custom_target(${project}NativeLLVMConfig DEPENDS ${${project}_LLVM_CONFIG_EXE})
-  add_dependencies(${project}NativeLLVMConfig CONFIGURE_LLVM_NATIVE)
-
-  add_dependencies(llvm-config ${project}NativeLLVMConfig)
-endif(CMAKE_CROSSCOMPILING)
+#if(CMAKE_CROSSCOMPILING)
+#  set(${project}_LLVM_CONFIG_EXE "${LLVM_NATIVE_BUILD}/bin/llvm-config")
+#  set(${project}_LLVM_CONFIG_EXE ${${project}_LLVM_CONFIG_EXE} PARENT_SCOPE)
+#
+#  add_custom_command(OUTPUT "${${project}_LLVM_CONFIG_EXE}"
+#    COMMAND ${CMAKE_COMMAND} --build . --target llvm-config --config $<CONFIGURATION>
+#    DEPENDS ${LLVM_NATIVE_BUILD}/CMakeCache.txt
+#    WORKING_DIRECTORY ${LLVM_NATIVE_BUILD}
+#    COMMENT "Building native llvm-config..."
+#    USES_TERMINAL)
+#  add_custom_target(${project}NativeLLVMConfig DEPENDS ${${project}_LLVM_CONFIG_EXE})
+#  add_dependencies(${project}NativeLLVMConfig CONFIGURE_LLVM_NATIVE)
+#
+#  add_dependencies(llvm-config ${project}NativeLLVMConfig)
+#endif(CMAKE_CROSSCOMPILING)
