diff --git a/xplat/hermes/external/llvh/include/llvh/ADT/DenseMap.h b/xplat/hermes/external/llvh/include/llvh/ADT/DenseMap.h
--- a/xplat/hermes/external/llvh/include/llvh/ADT/DenseMap.h
+++ b/xplat/hermes/external/llvh/include/llvh/ADT/DenseMap.h
@@ -1118,6 +1118,7 @@
   friend class DenseMapIterator<KeyT, ValueT, KeyInfoT, Bucket, false>;
 
   using ConstIterator = DenseMapIterator<KeyT, ValueT, KeyInfoT, Bucket, true>;
+  using NonConstIterator = DenseMapIterator<KeyT, ValueT, KeyInfoT, Bucket, false>;
 
 public:
   using difference_type = ptrdiff_t;
@@ -1176,12 +1177,18 @@
            "comparing incomparable iterators!");
     return Ptr == RHS.Ptr;
   }
-  bool operator!=(const ConstIterator &RHS) const {
+  bool operator==(const NonConstIterator &RHS) const {
     assert((!Ptr || isHandleInSync()) && "handle not in sync!");
     assert((!RHS.Ptr || RHS.isHandleInSync()) && "handle not in sync!");
     assert(getEpochAddress() == RHS.getEpochAddress() &&
            "comparing incomparable iterators!");
-    return Ptr != RHS.Ptr;
+    return Ptr == RHS.Ptr;
+  }
+  bool operator!=(const ConstIterator &RHS) const {
+    return !(*this == RHS);
+  }
+  bool operator!=(const NonConstIterator &RHS) const {
+    return !(*this == RHS);
   }
 
   inline DenseMapIterator& operator++() {  // Preincrement
