diff --git a/xplat/hermes/external/llvh/lib/Support/Windows/WindowsSupport.h b/xplat/hermes/external/llvh/lib/Support/Windows/WindowsSupport.h
--- a/xplat/hermes/external/llvh/lib/Support/Windows/WindowsSupport.h
+++ b/xplat/hermes/external/llvh/lib/Support/Windows/WindowsSupport.h
@@ -29,7 +29,9 @@
 // Require at least Windows 7 API.
 #define _WIN32_WINNT 0x0601
 #define _WIN32_IE    0x0800 // MinGW at it again. FIXME: verify if still needed.
+#ifndef WIN32_LEAN_AND_MEAN
 #define WIN32_LEAN_AND_MEAN
+#endif // !WIN32_LEAN_AND_MEAN
 #ifndef NOMINMAX
 #define NOMINMAX
 #endif
