diff --git a/external/llvh/cmake/modules/GetHostTriple.cmake b/external/llvh/cmake/modules/GetHostTriple.cmake
--- a/external/llvh/cmake/modules/GetHostTriple.cmake
+++ b/external/llvh/cmake/modules/GetHostTriple.cmake
@@ -14,6 +14,16 @@ function( get_host_triple var )
     else()
       set( value "i686-pc-windows-gnu" )
     endif()
+  elseif( CMAKE_HOST_WIN32 )
+    set( ABI_NORM "" )
+    if( ANDROID_ABI STREQUAL "armeabi-v7a" )
+      set( ABI_NORM "arm" )
+    elseif( ANDROID_ABI STREQUAL "arm64-v8a" )
+      set( ABI_NORM "arm64" )
+    else()
+      set( ABI_NORM "${ANDROID_ABI}" )
+    endif()
+    set( value "${ABI_NORM}-unknown-linux-gnu" )
   else( MSVC )
     set(config_guess ${LLVM_MAIN_SRC_DIR}/cmake/config.guess)
     execute_process(COMMAND sh ${config_guess}
