diff --git a/xplat/hermes/external/llvh/include/llvh/ADT/APInt.h b/xplat/hermes/external/llvh/include/llvh/ADT/APInt.h
--- a/xplat/hermes/external/llvh/include/llvh/ADT/APInt.h
+++ b/xplat/hermes/external/llvh/include/llvh/ADT/APInt.h
@@ -12,9 +12,6 @@
 /// integral constant values and operations on them.
 ///
 //===----------------------------------------------------------------------===//
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wshorten-64-to-32"
-
 #ifndef LLVM_ADT_APINT_H
 #define LLVM_ADT_APINT_H

@@ -25,6 +22,9 @@
 #include <cstring>
 #include <string>

+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wshorten-64-to-32"
+
 namespace llvh {
 class FoldingSetNodeID;
 class StringRef;
@@ -2211,5 +2211,6 @@
 hash_code hash_value(const APInt &Arg);
 } // End of llvm namespace

-#endif
 #pragma GCC diagnostic pop
+
+#endif
diff --git a/xplat/hermes/external/llvh/include/llvh/ADT/DenseMap.h b/xplat/hermes/external/llvh/include/llvh/ADT/DenseMap.h
--- a/xplat/hermes/external/llvh/include/llvh/ADT/DenseMap.h
+++ b/xplat/hermes/external/llvh/include/llvh/ADT/DenseMap.h
@@ -11,9 +11,6 @@
 //
 //===----------------------------------------------------------------------===//

-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wshorten-64-to-32"
-
 #ifndef LLVM_ADT_DENSEMAP_H
 #define LLVM_ADT_DENSEMAP_H

@@ -33,6 +30,9 @@
 #include <type_traits>
 #include <utility>

+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wshorten-64-to-32"
+
 namespace llvh {

 namespace detail {
@@ -1223,7 +1223,6 @@
 }

 } // end namespace llvh
+#pragma GCC diagnostic pop

 #endif // LLVM_ADT_DENSEMAP_H
-
-#pragma GCC diagnostic pop
diff --git a/xplat/hermes/external/llvh/include/llvh/ADT/DenseMapInfo.h b/xplat/hermes/external/llvh/include/llvh/ADT/DenseMapInfo.h
--- a/xplat/hermes/external/llvh/include/llvh/ADT/DenseMapInfo.h
+++ b/xplat/hermes/external/llvh/include/llvh/ADT/DenseMapInfo.h
@@ -11,9 +11,6 @@
 //
 //===----------------------------------------------------------------------===//

-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wshorten-64-to-32"
-
 #ifndef LLVM_ADT_DENSEMAPINFO_H
 #define LLVM_ADT_DENSEMAPINFO_H

@@ -26,6 +23,9 @@
 #include <cstdint>
 #include <utility>

+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wshorten-64-to-32"
+
 namespace llvh {

 template<typename T>
@@ -274,6 +274,6 @@

 } // end namespace llvh

-#endif // LLVM_ADT_DENSEMAPINFO_H
-
 #pragma GCC diagnostic pop
+
+#endif // LLVM_ADT_DENSEMAPINFO_H
diff --git a/xplat/hermes/external/llvh/include/llvh/ADT/Hashing.h b/xplat/hermes/external/llvh/include/llvh/ADT/Hashing.h
--- a/xplat/hermes/external/llvh/include/llvh/ADT/Hashing.h
+++ b/xplat/hermes/external/llvh/include/llvh/ADT/Hashing.h
@@ -41,10 +41,6 @@
 // under 32-bytes.
 //
 //===----------------------------------------------------------------------===//
-
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wshorten-64-to-32"
-
 #ifndef LLVM_ADT_HASHING_H
 #define LLVM_ADT_HASHING_H

@@ -58,6 +54,9 @@
 #include <string>
 #include <utility>

+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wshorten-64-to-32"
+
 namespace llvh {

 /// An opaque object representing a hash code.
@@ -660,6 +659,6 @@

 } // namespace llvh

-#endif
-
 #pragma GCC diagnostic pop
+
+#endif
diff --git a/xplat/hermes/external/llvh/include/llvh/ADT/MapVector.h b/xplat/hermes/external/llvh/include/llvh/ADT/MapVector.h
--- a/xplat/hermes/external/llvh/include/llvh/ADT/MapVector.h
+++ b/xplat/hermes/external/llvh/include/llvh/ADT/MapVector.h
@@ -13,8 +13,6 @@
 // a std::vector.
 //
 //===----------------------------------------------------------------------===//
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wshorten-64-to-32"
 #ifndef LLVM_ADT_MAPVECTOR_H
 #define LLVM_ADT_MAPVECTOR_H

@@ -28,6 +26,9 @@
 #include <utility>
 #include <vector>

+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wshorten-64-to-32"
+
 namespace llvh {

 /// This class implements a map that also provides access to all stored values
@@ -238,5 +239,6 @@

 } // end namespace llvh

-#endif // LLVM_ADT_MAPVECTOR_H
 #pragma GCC diagnostic pop
+
+#endif // LLVM_ADT_MAPVECTOR_H
diff --git a/xplat/hermes/external/llvh/include/llvh/ADT/SmallVector.h b/xplat/hermes/external/llvh/include/llvh/ADT/SmallVector.h
--- a/xplat/hermes/external/llvh/include/llvh/ADT/SmallVector.h
+++ b/xplat/hermes/external/llvh/include/llvh/ADT/SmallVector.h
@@ -11,9 +11,6 @@
 //
 //===----------------------------------------------------------------------===//

-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wshorten-64-to-32"
-
 #ifndef LLVM_ADT_SMALLVECTOR_H
 #define LLVM_ADT_SMALLVECTOR_H

@@ -36,6 +33,9 @@
 #include <type_traits>
 #include <utility>

+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wshorten-64-to-32"
+
 namespace llvh {

 /// This is all the non-templated stuff common to all SmallVectors.
@@ -944,6 +944,5 @@

 } // end namespace std

-#endif // LLVM_ADT_SMALLVECTOR_H
-
 #pragma GCC diagnostic pop
+#endif // LLVM_ADT_SMALLVECTOR_H
diff --git a/xplat/hermes/external/llvh/include/llvh/ADT/StringRef.h b/xplat/hermes/external/llvh/include/llvh/ADT/StringRef.h
--- a/xplat/hermes/external/llvh/include/llvh/ADT/StringRef.h
+++ b/xplat/hermes/external/llvh/include/llvh/ADT/StringRef.h
@@ -6,9 +6,6 @@
 // License. See LICENSE.TXT for details.
 //
 //===----------------------------------------------------------------------===//
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wshorten-64-to-32"
-
 #ifndef LLVM_ADT_STRINGREF_H
 #define LLVM_ADT_STRINGREF_H

@@ -24,6 +21,9 @@
 #include <type_traits>
 #include <utility>

+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wshorten-64-to-32"
+
 namespace llvh {

   class APInt;
@@ -936,5 +936,5 @@

 } // end namespace llvh

-#endif // LLVM_ADT_STRINGREF_H
 #pragma GCC diagnostic pop
+#endif // LLVM_ADT_STRINGREF_H
diff --git a/xplat/hermes/external/llvh/include/llvh/IR/InstrTypes.h b/xplat/hermes/external/llvh/include/llvh/IR/InstrTypes.h
--- a/xplat/hermes/external/llvh/include/llvh/IR/InstrTypes.h
+++ b/xplat/hermes/external/llvh/include/llvh/IR/InstrTypes.h
@@ -12,9 +12,6 @@
 // i*.h files...
 //
 //===----------------------------------------------------------------------===//
-#pragma GCC diagnostic push
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wshorten-64-to-32"
 #ifndef LLVM_IR_INSTRTYPES_H
 #define LLVM_IR_INSTRTYPES_H

@@ -45,6 +42,9 @@
 #include <string>
 #include <vector>

+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wshorten-64-to-32"
+
 namespace llvh {

 //===----------------------------------------------------------------------===//
@@ -1492,6 +1492,5 @@
 };

 } // end namespace llvh
-
-#endif // LLVM_IR_INSTRTYPES_H
 #pragma GCC diagnostic pop
+#endif // LLVM_IR_INSTRTYPES_H
diff --git a/xplat/hermes/external/llvh/include/llvh/Support/MathExtras.h b/xplat/hermes/external/llvh/include/llvh/Support/MathExtras.h
--- a/xplat/hermes/external/llvh/include/llvh/Support/MathExtras.h
+++ b/xplat/hermes/external/llvh/include/llvh/Support/MathExtras.h
@@ -11,9 +11,6 @@
 //
 //===----------------------------------------------------------------------===//

-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wshorten-64-to-32"
-
 #ifndef LLVM_SUPPORT_MATHEXTRAS_H
 #define LLVM_SUPPORT_MATHEXTRAS_H

@@ -26,6 +23,9 @@
 #include <limits>
 #include <type_traits>

+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wshorten-64-to-32"
+
 #ifdef __ANDROID_NDK__
 #include <android/api-level.h>
 #endif
@@ -853,7 +853,5 @@
 /// Use this rather than HUGE_VALF; the latter causes warnings on MSVC.
 extern const float huge_valf;
 } // End llvm namespace
-
-#endif
-
 #pragma GCC diagnostic pop
+#endif
