diff --git a/external/llvh/cmake/modules/GetHostTriple.cmake b/external/llvh/cmake/modules/GetHostTriple.cmake
--- a/external/llvh/cmake/modules/GetHostTriple.cmake
+++ b/external/llvh/cmake/modules/GetHostTriple.cmake
@@ -14,6 +14,9 @@ function( get_host_triple var )
     else()
       set( value "i686-pc-windows-gnu" )
     endif()
+  elseif(CMAKE_HOST_WIN32)
+    message(WARNING "LLVM_HOST_TRIPLE can't be computed on Windows with this toolchain")
+    set( value "unknown-unknown-unknown" )
   else( MSVC )
     set(config_guess ${LLVM_MAIN_SRC_DIR}/cmake/config.guess)
     execute_process(COMMAND sh ${config_guess}
