diff --git a/external/llvh/include/llvh/Support/Timer.h b/external/llvh/include/llvh/Support/Timer.h
--- a/external/llvh/include/llvh/Support/Timer.h
+++ b/external/llvh/include/llvh/Support/Timer.h
@@ -175,6 +175,7 @@
     std::string Description;
 
     PrintRecord(const PrintRecord &Other) = default;
+    PrintRecord &operator=(const PrintRecord &Other) = default;
     PrintRecord(const TimeRecord &Time, const std::string &Name,
                 const std::string &Description)
       : Time(Time), Name(Name), Description(Description) {}
